src/PointwiseFunctions/AnalyticSolutions/Elasticity/BentBeam.hpp

```diff
@@ -0,0 +1,144 @@
+
+#include "DataStructures/DataBox/Prefixes.hpp"   // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"      // IWYU pragma: keep
+#include "Elliptic/Systems/Elasticity/Tags.hpp"  // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/IsotropicHomogeneous.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Elasticity {
+namespace Solutions {
+/*!
+ * \brief A state of pure bending of an elastic beam in 2D
+ *
+ * \details This solution describes a 2D slice through an elastic beam of length
+ * \f$L\f$ and height \f$H\f$ that is subject to a bending moment \f$M\f$. The
+ * beam material is characterized by an isotropic and homogeneous constitutive
+ * relation \f$Y^{ijkl}\f$ in the plane-stress approximation (see
+ * `Elasticity::ConstitutiveRelations::IsotropicHomogeneous`). In this scenario,
+ * no body-forces \f$f_\mathrm{ext}^j\f$ act on the material, so the
+ * \ref Elasticity equations reduce to \f$\nabla_i T^{ij}=0\f$, but the bending
+ * moment \f$M\f$ generates the stress \f[
+ * T^{xx} = \frac{3M}{2h^3} y \\
+ * T^{xy} = 0 = T^{yy} \text{.}
+ * \f]
+ * By fixing the rigid-body motions to
+ * \f[
+ * u^x(0,y)=0 \quad \text{and} \quad u^y(\pm L,0)=0
+ * \f]
+ * we find that this stress is produced by the displacement field
+ * \f{align}
+ * u^x&=-\frac{3M}{2Eh^3}xy
+ * u^y&=\frac{3M}{4Eh^3}(x^2+\nu y^2-L^2)
+ * \f}
+ * in terms of the Young's modulus \f$E\f$ and the Poisson ration \f$\nu\f$ of
+ * the material.
+ */
+class BentBeam {
+ public:
+  using constitutive_relation_type =
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<2>;
+
+  struct Length {
+    using type = double;
+    static constexpr OptionString help{"The beam length"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Height {
+    using type = double;
+    static constexpr OptionString help{"The beam height"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct BendingMoment {
+    using type = double;
+    static constexpr OptionString help{
+        "The bending moment applied to the beam"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  using options = tmpl::list<
+      Length, Height, BendingMoment,
+      Elasticity::Tags::ConstitutiveRelation<constitutive_relation_type>>;
+  static constexpr OptionString help{
+      "An elastic beam subject to a bending moment in 2D"};
+
+  BentBeam() = default;
+  BentBeam(const BentBeam&) noexcept = delete;
+  BentBeam& operator=(const BentBeam&) noexcept = delete;
+  BentBeam(BentBeam&&) noexcept = default;
+  BentBeam& operator=(BentBeam&&) noexcept = default;
+  ~BentBeam() noexcept = default;
+
+  explicit BentBeam(double length, double height, double bending_moment,
```

Comment:
my bad, no need for the `explicit` here since the constructor takes multiple arguments. If it were only one argument we'd need `explicit` to avoid unintentional conversion from the constructor argument (e.g. `BentBeam solution = 1.`).